### PR TITLE
Update scaffolding bats integration tests

### PR DESCRIPTION
**Pull request by c12i**
### TODO:
- [ ] CHANGELOGs updated with appropriate info

### CODE REVIEW

**Comment by c12i** [comment]
It appears that [this function](https://github.com/holochain/scaffolding/blob/d178658efec8c006b71a8929474111b6d27432a5/src/scaffold/app/nix.rs#L70) call from [here](https://github.com/holochain/scaffolding/blob/d178658efec8c006b71a8929474111b6d27432a5/src/cli.rs#L904) is resulting in this error: `IO error: Read-only file system (os error 30)`.

**Comment by c12i** [comment]
@ThetaSinner and @steveej, running into two issues here.

It appears that adding [`extra-experimental-features = flakes nix-command`](https://github.com/holochain/scaffolding/blob/d178658efec8c006b71a8929474111b6d27432a5/src/scaffold/app/nix.rs#L110) seems to fail in the runner, causing the scaffolded folder to be deleted via a `fs::remove_dir_all(&app_dir)?` call that is causing the [`IO error: Permission denied (os error 13)`](https://github.com/holochain/holochain/actions/runs/8882417459/job/24387060029?pr=3761#step:4:148) leading to the ci job failures

File: holonix/test/scaffolding.bats [hunk 1]
```diff
@@ -18,5 +18,24 @@ teardown() {
 
 @test "expected scaffold an example to succeed" {
   set -e
-  hc scaffold example forum -t vue
```
[the remark this answers is not in the document]
Comment (steveej) [reply holonix/test/scaffolding.bats:21]: @c12i i think this needs to be fixed in scaffolding. are you going to take it from here or should i create an issue?

[the remark this answers is not in the document]
Comment (c12i) [reply holonix/test/scaffolding.bats:21]: Thanks for catching this, I know what the issue is, the arguments are a little different with different scaffolding versions. I can look into how I can fix this.

**Comment by steveej** [comment]
@c12i is this effort still interesting to you or has an alternative way for integration testing this emerged?

**Comment by c12i** [comment]
ci jobs seem to get stuck for some reason @steveej

File: holonix/test/scaffolding.bats (hunk 1 above)
[the remark this answers is not in the document]
Comment (c12i) [reply holonix/test/scaffolding.bats:21]: Addressed the initial cause of the failure, which was due to differences between how scaffolding generally works with regards to cli inputs between `0.1`/ `0.2` and `0.3`/`0.4`. Still running into the `IO Error` though, so I believe the missing `nix` binary still needs to be added.

**Pull request closed without merging**